pkg/sidecar/k8s_instance.go

```diff
@@ -454,3 +459,36 @@ func getRedisRoute(handle *netlink.Handle, redisIP net.IP) (*netlink.Route, erro
 
 	return &redisRoute, nil
 }
+
+func waitForPodRunningPhase(ctx context.Context, podName string) error {
+	k8scfg, err := clientcmd.BuildConfigFromFlags("", "")
+	if err != nil {
+		return fmt.Errorf("error in wait for pod running phase: %v", err)
+	}
+
+	k8sClientset, err := kubernetes.NewForConfig(k8scfg)
+	if err != nil {
+		return fmt.Errorf("error in wait for pod running phase: %v", err)
+	}
+
+	var phase string
+
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("wait for pod context (pod name: %s) erred with: %w", podName, ctx.Err())
+		default:
+			if phase == "Running" {
+				return nil
+			}
+			pod, err := k8sClientset.CoreV1().Pods("default").Get(podName, metav1.GetOptions{})
+			if err != nil {
+				return fmt.Errorf("error in wait for pod running phase: %v", err)
+			}
+
+			phase = string(pod.Status.Phase)
+
+			time.Sleep(1 * time.Second)
```

Comment:
This should probably select on a context.

Reply:
Really, we could replace the `default:` with a ticker/timer.

Reply:
I think this is simpler, than having a ticker that has to instantly tick (as we don't want to wait on the first entry of this case), and then have to `Timer.Reset` it to the period (1 second in this case)... just time.Sleep seems much simpler, no?

Reply:
The benefit of the ticker is that we won't be delaying the cancelation if it happens though...